### ISSUE TITLE
[useRequest] `data` must not be undefined if `initialData` is set

### ISSUE THREAD

**Issue by nikolay-borzov** [alibaba/hooks#1025]
There is a TypeScript typing error.

When calling `useRequest` with `initialData` is set the `data` must be not undefined

```ts
const getList = async (id: number): string[] => ["one", "two", "three"];

const { data: options } = useRequest(
    () => getList(id),
    {
      refreshDeps: [id],
      initialData: [],
    },
  );

// Object is possibly 'undefined'
options.forEach(option => { console.log(option) })
```

Since `initialData` is set, `options` has to be equal to `[]`

**Comment by brickspert** [comment]
Yes, you are right. 

Now you can write like this to resolve is: 

```ts
const { data: options = [] } = useRequest(
    () => getList(id),
    {
      refreshDeps: [id]
    },
  );
```

We will fix this in next major version.

**Comment by nikolay-borzov** [comment]
It seems more like a patch (0.0.1) to me. Why does it require a major version change?

**Comment by brickspert** [comment]
We have removed initialData in 3.0~



Hello：

这个问题我们已经在 v3.0 版本彻底修复，目前 v3.0 版本已经发布 alpha 版本。

欢迎试用。

文档：https://ahooks-next.surge.sh/
源码：https://github.com/alibaba/hooks/tree/next

安装最新分支
```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```

We have fixed this problem in the v3.0 version, and the alpha has been released. 

Welcome to try.

Doc：https://ahooks-next.surge.sh/
Source Code：https://github.com/alibaba/hooks/tree/next

Install v3.0

```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```
